src/Symfony/Component/Config/Resource/ResourceInterface.php

```diff
@@ -31,6 +37,9 @@ public function __toString();
      * @param int $timestamp The last time the resource was loaded
      *
      * @return bool True if the resource has not been updated, false otherwise
+     *
+     * @deprecated since 2.8, to be removed in 3.0. If your resource can check itself for
+     *             freshness implement the SelfCheckingResourceInterface instead.
```

Comment:
If they need service-based validation, they will have their own checkers which will be run first. In that case, they can implement the method but leave it empty and it will not even be called. 

In 3.0, they can remove the method.

I don't see any other way to get this working BC.